engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/CephSnapshotStrategy.java

```diff
@@ -64,10 +64,6 @@ public StrategyPriority canHandle(Snapshot snapshot, SnapshotOperation op) {
             return StrategyPriority.HIGHEST;
         }
 
-        if (SnapshotOperation.DELETE.equals(op)) {
-            return StrategyPriority.HIGHEST;
-        }
-
```

Comment:
code LGTM, as delete op is not handled here, it shouldn't consider this strategy.

Reply:
@sureshanaparti thanks for the code review. I would like to just add a note to avoid misunderstandings.

Delete operation for Ceph has always been supported by CephSnapshotStrategy due to the fact that the _parent_ class of CephSnapshotStrategy is StorageSystemSnapshotStrategy, which handles the deletion of Ceph/RBD snapshots.

However, StorageSystemSnapshotStrategy does not handle backed snapshots via NFS in this context and when Ceph had features introduced the backup on secondary storage was not taken into account.

With the fix proposed by @slavkap, it was possible to check that **nowadays** DefaultSnapshotStrategy supports deletions of Snapshots on Ceph (with or without backup on NFS).